src/expr/impl/src/scalar/jsonb_access.rs

```diff
@@ -39,6 +69,54 @@ pub fn jsonb_array_element(v: JsonbRef<'_>, p: i32) -> Option<JsonbRef<'_>> {
     v.access_array_element(idx)
 }
 
+/// Extracts JSON sub-object at the specified path, where path elements can be either field keys or array indexes.
+///
+/// `jsonb #> text[] → jsonb`
+///
+/// # Examples
+///
+/// ```slt
+/// query T
+/// select '{"a": {"b": ["foo","bar"]}}'::jsonb #> '{a,b,1}'::text[];
+/// ----
+/// "bar"
+///
+/// query T
+/// select '{"a": {"b": ["foo","bar"]}}'::jsonb #> '{a,b,null}'::text[];
+/// ----
+/// NULL
+/// ```
+#[function("jsonb_access_multi(jsonb, varchar[]) -> jsonb")]
+pub fn jsonb_access_multi<'a>(v: JsonbRef<'a>, path: ListRef<'_>) -> Option<JsonbRef<'a>> {
+    let mut jsonb = v;
+    for key in path.iter() {
+        // return null if any element is null
+        let key = key?.into_utf8();
+        jsonb = match key.parse() {
+            Ok(idx) => jsonb_array_element(jsonb, idx)?,
+            Err(_) => jsonb_object_field(jsonb, key)?,
+        };
```

Comment:
Based on the regress test failure, it seems we should choose whether to index by number (`parse`) or string based on jsonb type, rather than the reverse.